packages/integration-tests/src/e2e-flow/setup_package_manager.ts

```
@@ -0,0 +1,120 @@
+  await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+  await execa(
+    'yarn',
+    ['config', 'set', 'registry', 'http://localhost:4873'],
+    execaOptions
+  );
+  await execa('yarn', ['config', 'get', 'registry'], execaOptions);
+  await execa('yarn', ['cache', 'clean'], execaOptions);
+};
+
+const initializeYarnModern = async (execaOptions: {
+  cwd: string;
+  stdio: 'inherit';
+}) => {
+  {
+    await execa('npm', ['install', '-g', 'yarn'], { stdio: 'inherit' });
+    await execa('yarn', ['init', '-2'], execaOptions);
+    await execa(
+      'yarn',
+      ['config', 'set', 'npmRegistryServer', 'http://localhost:4873'],
+      execaOptions
+    );
+    await execa(
+      'yarn',
+      ['config', 'set', 'unsafeHttpWhitelist', 'localhost'],
+      execaOptions
+    );
+    await execa(
+      'yarn',
+      ['config', 'set', 'nodeLinker', 'node-modules'],
+      execaOptions
+    );
+    await execa('yarn', ['cache', 'clean'], execaOptions);
+  }
+};
+
+/**
+ * Sets up the package manager for the e2e flow
+ */
+export const setupPackageManager = async (
+  dir: string
+): Promise<{
+  packageManager: PackageManager;
+  packageManagerExecutable: PackageManagerExecutable;
+}> => {
+  const packageManager = (process.env.PACKAGE_MANAGER ??
+    'npm') as PackageManager;
+  const packageManagerExecutable = packageManager.startsWith('yarn')
+    ? 'yarn'
+    : packageManager === 'npm'
+    ? 'npx'
+    : (packageManager as PackageManagerExecutable);
+  const execaOptions = {
+    cwd: os.homedir(),
+    stdio: 'inherit' as const,
+  };
+
+  switch (packageManager) {
+    case 'npm':
+      await initializeNpm();
+      break;
+
+    case 'pnpm':
+      await initializePnpm(packageManager);
+      break;
+
+    case 'yarn-classic':
+      await initializeYarnClassic(execaOptions);
+      break;
+
+    case 'yarn-modern':
+      execaOptions.cwd = dir;
+      await initializeYarnModern(execaOptions);
+      break;
+
+    default:
+      break;
```

we should probably throw an error here

---

Done 60d33bc179